### PR TITLE
MULE-11875: Race condition when putting an object in the registry

### PR DESCRIPTION
**Pull request by elrodro83**
asynchronously and disposing the muleContext at the same time

Also includes:
* Fix for the issue that prevented running integration tests alone
* Name of the lifecycle manager was not being properly generated

### CODE REVIEW

File: modules/spring-config/src/test/java/org/mule/runtime/config/spring/SpringRegistryTestCase.java [hunk 1]
```diff
@@ -71,4 +90,65 @@ private void createSpringRegistryWithParentContext() {
...
+    springRegistry.initialise();
+    springRegistry.getLifecycleManager().fireLifecycle(Startable.PHASE_NAME);
+
+    final Startable asyncStartableBean = mock(Startable.class, withSettings().extraInterfaces(Disposable.class));
+
+    final Latch startingLatch = new Latch();
+
+    doAnswer(invocation -> {
+      startingLatch.countDown();
+      sleep(1000);
```
Comment (marianogonzalez) [inline modules/spring-config/src/test/java/org/mule/runtime/config/spring/SpringRegistryTestCase.java]: this sleep should be replaced with a second latch. This is a potential source of flakyness

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/LifecycleAwareConfigurationProvider.java [hunk 2]
```diff
@@ -56,8 +58,7 @@
   private final ConfigurationModel configurationModel;
   private final List<ConfigurationInstance> configurationInstances = new LinkedList<>();
   private final ClassLoader extensionClassLoader;
-  protected SimpleLifecycleManager lifecycleManager =
-      new DefaultLifecycleManager<>(String.format("%s-%s", getClass().getName(), getName()), this);
+  protected final SimpleLifecycleManager lifecycleManager;
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/LifecycleAwareConfigurationProvider.java]: what do you gain from this change?

File: tests/integration/pom.xml [hunk 3]
```diff
@@ -92,6 +92,7 @@
             <groupId>org.mule.modules</groupId>
             <artifactId>mule-module-file-extension-common</artifactId>
             <version>${project.version}</version>
+            <classifier>mule-plugin</classifier>
```
Comment (marianogonzalez) [inline tests/integration/pom.xml]: how is this related?

File: modules/spring-config/src/test/java/org/mule/runtime/config/spring/SpringRegistryTestCase.java (hunk 1 above)
Comment (elrodro83) [inline modules/spring-config/src/test/java/org/mule/runtime/config/spring/SpringRegistryTestCase.java]: a latch is for waiting for sonething to happen. here i have to check that something has not happenned after some time.

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/LifecycleAwareConfigurationProvider.java (hunk 2 above)
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/LifecycleAwareConfigurationProvider.java]: when this was being executed, the name was not set still and the string was ended with `-null`

File: tests/integration/pom.xml (hunk 3 above)
Comment (elrodro83) [inline tests/integration/pom.xml]: i needed to fix this in order to run the integration tests and analyze it.

File: modules/spring-config/src/test/java/org/mule/runtime/config/spring/SpringRegistryTestCase.java (hunk 1 above)
Comment (marianogonzalez) [inline modules/spring-config/src/test/java/org/mule/runtime/config/spring/SpringRegistryTestCase.java]: "waiting for some time" is the definition of flaky. In the bug you're trying to fix, why was this object being wrongfully disposed? what was triggering that disposal? You can add a second latch which is fired by that condition and THEN you verify that it was never disposed.
